services/sentry/lib/eventReducer.js

```diff
@@ -6,9 +6,9 @@ function newAggr(channelId) {
 	return { channelId, created: new Date(), events: {}, totals: {}, earners: [] }
 }
 
-function reduce(channel, initialAggr, ev) {
+function reduce(channel, session, initialAggr, event) {
 	let aggr = { ...initialAggr }
-
+	const ev = { ...event, ...session } // add session details (i.e country, device type etc) to event
```

Comment:
why don't you just pass the `session` to `getPayout`

Reply:
please answer this @samparsky

Reply:
i have changed it to pass `session`